packages/rudy/src/middleware/codeSplit.js

```diff
@@ -0,0 +1,43 @@
+import { createStore, applyMiddleware, compose, combineReducers } from 'redux'
+
+import { tryRequire } from '../../AsyncComponent/index'
+
+export default (name = 'load') => (api) => async (req, next) => {
+  const load = req.route && req.route[name]
```

Comment:
For context, this is kind of the way I've been thinking of implementing code splitting:
```javascript
{
  [ASYNC_ROUTE]: {
    //...
    beforeMaybeEnter: async (dispatch, getState, bag) => {
       // This is my favourite way to do reducer splitting in RFR
      const { req, extra: { addReducer } } = bag;

      // Standard format object with things to load like AsyncComponent here
      const { reducers, components, loadData } = await import('./routeThunks/asyncRoute');
      // components is not used and probably doesn't need to be returned, but it should
      // be imported so that when we commit the route the code is ready

      // This part is very boilerplate, the same for every route
      Object.keys(reducers).forEach((name) => addReducer(name, reducers[name]));

      /**
       * This is like, the rest of the thunk. I'm assuming here that its not a problem to execute
       * this thunk even though there might still be a redirect, or the user might not actually
       * click on the link - the assumption is that it won't actually change the UI, but it will hopefully
       * get everything ready (including loading data from the API) so that when they do it will be
       * instant (or at least faster)
       *
       * Probably inside load() there are some of the same import() calls as there are in
       * 'routeThunks/asyncRoute', but at this point we trust that they will resolve without
       * downloading any code.
       */
      return load(dispatch, getState, bag);
    },
    afterEnter: (dispatch, getState, bag) => {
       // Now the route is definitely happening and there is nothing anyone can do about it.
       recordPageLoadForAnalytics(dispatch, getState, bag);
       triggerReallyExpensiveApiCallIDontWantToCache(dispatch, getState, bag);
       doSomethingWithSideEffectsThatCantBePreloaded(dispatch, getState, bag);
    },
  },
}
```